p4c_bm/gen_json.py

```diff
@@ -1198,6 +1210,30 @@ def dump_field_aliases(json_dict, hlir, path_field_aliases):
     field_aliases = [[a, v] for a, v in aliases_dict.items()]
     json_dict["field_aliases"] = field_aliases
 
+def dump_extern_instances(json_dict, hlir):
+    extern_instances = []
+    id_ = 0
+    for name, p4_extern_instance in hlir.p4_extern_instances.items():
+        extern_instance_dict = OrderedDict()
+        extern_instance_dict["name"] = name
+        extern_instance_dict["id"] = id_
+        extern_instance_dict["type"] = p4_extern_instance.extern_type.name
+
+        id_ += 1
+
+        attributes = []
+        for attribute, attr in p4_extern_instance.attributes.items():
+            attribute_dict = OrderedDict()
+            attribute_dict["name"] = attribute
+            attribute_dict["type"] = "hexstr" # Ref: P4Objects.cpp, line 264
+            attribute_dict["value"] = attr.strip()
```

Comment:
If you look at this file (https://github.com/p4lang/p4c-bm/blob/master/tox.ini), you will see that both requirements (v1 and v1.1) are installed. So you don't need to worry about dependencies here. If you add a test program under `tests/p4_programs/v1_1/` it will be run using the v1.1 HLIR.

Regarding the attribute type, it would be nice if you could change it to an integer value, as it is the only thing bmv2 actually supports. In your P4 program you can change the attribute type to `bit<1>`?